src/Symfony/Component/Messenger/Bridge/Redis/Transport/Connection.php

```diff
@@ -84,6 +86,7 @@ public function __construct(array $configuration, array $connectionCredentials =
         $this->autoSetup = $configuration['auto_setup'] ?? self::DEFAULT_OPTIONS['auto_setup'];
         $this->maxEntries = $configuration['stream_max_entries'] ?? self::DEFAULT_OPTIONS['stream_max_entries'];
         $this->deleteAfterAck = $configuration['delete_after_ack'] ?? self::DEFAULT_OPTIONS['delete_after_ack'];
+        $this->deleteAfterReject = $configuration['delete_after_reject'] ?? $configuration['delete_after_ack'] ?? self::DEFAULT_OPTIONS['delete_after_reject'];
```

Comment:
I'm not sure binding delete_after_reject to delete_after_ack is expected.
It would be less surprising to both options independent to me:
`$configuration['delete_after_reject'] ?? self::DEFAULT_OPTIONS['delete_after_reject'];`

Reply:
My reasoning for this is:  
If someone is already using `delete_after_ack` for reducing memory usage of Redis, then they would expect rejection to work similarly.

But maybe I'm just overthinking this :smile:

Reply:
I think we should let ppl be able to decide, there will be an option anyway. Their choice.

Reply:
I agree with @nicolas-grekas here

Reply:
@chalasr @nicolas-grekas changes applied as suggested.